pkg/secrets/check_rights_nix.go

```diff
@@ -90,3 +91,40 @@ func checkGroupPermission(stat *syscall.Stat_t, usr *user.User, userGroups []str
 
 	return nil
 }
+
+var checkConfigRights = func(path string) error {
```

Comment:
nit: The newly added functions are not in the style of the rest of the file's functions.
nit: We also may want to name this according to it's function (e.g. `ensureRootOwnership` or something similar)

Reply:
> nit: The newly added functions are not in the style of the rest of the file's functions.
Thanks, mind providing some specifics? Missing comments? Name? I am struggling with the name and the `ensureRootOwnership` is not representative of both *nix and Windows cases - maybe `ensureConfigRightsRestricted`? I am following the style of `checkRights` but I need to stub this for tests in a cross platform way. Hence the `= func()`.

Reply:
For the style, I would expect them to match the others with `func foo(...) {` rather than `var foo = func(...) {`. If as you mention we're doing it for stubbing, I've seen the [dependency injection pattern](https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/worker/worker.go#L45-L80C27) work pretty well for most cases and it usually results in cleaner code with less bugs since the tests change the minimum needed (though sometimes it requires code reorg).

For the name, you make a good point - maybe `ensureConfigFilePermissions`?

Reply:
I somewhat agree re: dependencies injection, going to defer it to consider refactoring later as it will result in more changes to this already rather complex PR.

I can change to `ensureConfigFilePermissions` or `ensureConfigFilePermissionsRestricted` but these are not in the style of naming in this file. 😄 

There is a precedent of using initialized stubs in this package [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/secrets/secrets.go#L127-L128) and [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/secrets/fetch_secret.go#L112-L113)

Reply:
> I somewhat agree re: dependencies injection, going to defer it to consider refactoring later as it will result in more changes to this already rather complex PR. There is a precedent of using initialized stubs in this package [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/secrets/secrets.go#L127-L128) and [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/secrets/fetch_secret.go#L112-L113)

Fair enough though I would say that even if there's precedent, we should try to improve things wherever we can and especially in newly modified code.

> I can change to ensureConfigFilePermissions or ensureConfigFilePermissionsRestricted but these are not in the style of naming in this file. 😄

`checkConfigFilePermissions`?